### DISCUSSION TITLE
Issue with AWS CloudFormation CreateStack Operator Parameter Validation

### DISCUSSION THREAD

**Discussion by adarshsaraogi** [apache/airflow#37185] [General]
### Apache Airflow version

main (development)

### If "Other Airflow 2 version" selected, which one?

2.7.2

### What happened?

I hope this message finds you well. I wanted to bring to your attention an issue we've encountered while using the AWS provider with the CloudFormation CreateStack operator.Upon attempting to create a stack, we consistently encounter a validation error related to stack parameters. We've thoroughly reviewed our configurations and confirmed there are no apparent issues on our end. It seems there might be a core logic problem within the code.Unfortunately, due to corporate restrictions, I'm unable to share the entire code. However, I can provide a sample configuration along with the specific error. Could you please investigate this matter from your end?

### What you think should happen instead?

It should create a stack in Cloud formation.

### How to reproduce

```console 
CLOUDFORMATION_TEMPLATE = { "Description": "Stack from Airflow CloudFormation example DAG", "Resources": { "ExampleQueue": { "Type": "AWS::SQS::Queue", } }, } 
```

```python
task_replication_instance_setup = CloudFormationCreateStackOperator( task_id="smple", stack_name="DMS_Replication_Instances_Launch", cloudformation_parameters=json.dumps(cloudformation_create_parameters ))
```

### Error

Invalid type for parameter TemplateBody, value: {'Description': 'Stack from Airflow CloudFormation example DAG', 'Resources': {'ExampleQueue': {'Type':AWS::SQS::Queue'}}}, type: <class 'dict'>, valid types: <class 'str'>

### Operating System

MWAA

### Deployment

Amazon (AWS) MWAA

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
[Dashboard](https://aws-mwaa.github.io/#/open-source/system-tests/dashboard.html) provided by Amazon (AWS) Team shows that system tests with CloudFormation operators works fine.

So seems like this is a troubleshooting rather than a bug, convert into the discussion.
